src/Module.zig

```diff
@@ -3703,6 +3703,44 @@ fn semaDecl(mod: *Module, decl: *Decl) !bool {
         };
     };
 
+    if (mod.comp.verbose_air) {
+        // Duplicate everything so that we can make modifications. It doesn't
+        // look like the rest of this fn uses this so we could modify it in
+        // place but this makes it more future-proof to any semaDecl changes.
+        var air_extra = try sema.air_extra.clone(gpa);
+        var air_instructions = try sema.air_instructions.clone(gpa);
+        var air_values = try sema.air_values.clone(gpa);
+        defer air_extra.deinit(gpa);
+        defer air_instructions.deinit(gpa);
+        defer air_values.deinit(gpa);
+
+        // Copy the block into place and mark that as the main block. This is
+        // required for print_air to find the "main" block of the AIR.
+        try air_extra.ensureUnusedCapacity(gpa, @typeInfo(Air.Block).Struct.fields.len +
+            block_scope.instructions.items.len);
+        const main_block_index = @intCast(u32, air_extra.items.len);
+        air_extra.appendAssumeCapacity(@intCast(u32, block_scope.instructions.items.len));
+        air_extra.appendSliceAssumeCapacity(block_scope.instructions.items);
+        air_extra.items[@enumToInt(Air.ExtraIndex.main_block)] = main_block_index;
+
+        var air = Air{
+            .instructions = air_instructions.toOwnedSlice(),
+            .extra = air_extra.toOwnedSlice(gpa),
+            .values = air_values.toOwnedSlice(gpa),
+        };
+        defer air.deinit(gpa);
+
+        var liveness = try @import("Liveness.zig").analyze(gpa, air, zir);
+        defer liveness.deinit(gpa);
```

Comment:
We don't really want to analyze Liveness in this case - doing it just so that we can print the AIR is a bit misleading and is triggering code paths that were never meant to be covered.

I get that this would then require a bunch more effort to make AIR printable without liveness information attached to it.